apps/webshop/models.py

```diff
@@ -23,6 +23,22 @@ class Product(models.Model):
     deadline = models.DateTimeField(null=True, blank=True)
     active = models.BooleanField(default=True)
 
+    def calculate_stock(self):
+        if self.product_sizes.count() > 0:
+            return sum((product_size.stock for product_size in self.product_sizes.all()))
+        else:
+            return self.stock
+
+    def in_stock(self):
+        stock = self.calculate_stock()
+        return stock is None or stock > 0
+
+    def enough_stock(self, quantity, product_size=None):
+        stock = self.stock
+        if product_size:
+            stock = product_size.stock
+        return stock is None or stock >= quantity
```

Comment:
If stock isn't specified then it is assumed that there is an unlimited supply yes.